src/main/java/training/bank/Account.java

```diff
@@ -1,30 +1,34 @@
-package training.supportbank;
+package training.bank;
 
+import java.lang.reflect.Array;
+import java.math.BigDecimal;
+import java.util.ArrayList;
 import java.util.HashSet;
+import java.util.List;
 
 public class Account {
     private String owner;
-    private double balance;
+    private BigDecimal balance = BigDecimal.ZERO;
     private HashSet<Transaction> debts = new HashSet<>();
     private HashSet<Transaction> credits = new HashSet<>();
 
     public Account(String owner) {
         this.owner = owner;
     }
 
-    public void increaseBalance(double amount) {
-        balance += amount;
+    public void increaseBalance(BigDecimal amount) {
+        balance = balance.add(amount);
     }
 
-    public void decreaseBalance(double amount) {
-        balance -= amount;
+    public void decreaseBalance(BigDecimal amount) {
```

Comment:
I don't think it makes sense to update the balance without having added a transaction, and doesn't make sense to add a transaction without updating the balance. There should probably be a single method that does both, and that prevents this class from being misused.